File: 28-FindTheIndexOfTheFirstOccurrenceInAString/28-FindTheIndexOfTheFirstOccurrenceInAString.py
# Last updated: 20/07/2026, 18:40:24
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        # return haystack.find(needle)
        n=len(needle)
        for i in range(len(haystack)-len(needle)+1):
            if haystack[i:i+n]==needle:
                return i
        return -1
        
        

File: 28-FindTheIndexOfTheFirstOccurrenceInAString/28-FindTheIndexOfTheFirstOccurrenceInAString.java
// Last updated: 20/07/2026, 18:40:29
class Solution {
    public int strStr(String haystack, String needle) {
        if(needle.isEmpty())
        {
            return 0;
        }
        else
        {
            return(haystack.indexOf(needle));
        }
    }
}